include/sys/cbprintf.h

```diff
@@ -49,22 +49,25 @@ extern "C" {
  * @{
  */
 
-/** @brief Required alignment of the buffer used for packaging. */
+#define POWER_OF_TWO(x) ((x) && !((x) & ((x) - 1)))
+
 #ifdef __xtensa__
-#define CBPRINTF_PACKAGE_ALIGNMENT 16
-#elif defined(CONFIG_X86) && !defined(CONFIG_64BIT)
-/* sizeof(long double) is 12 on x86-32, which is not power of 2.
- * So set it manually.
- */
-#define CBPRINTF_PACKAGE_ALIGNMENT \
-	(IS_ENABLED(CONFIG_CBPRINTF_PACKAGE_LONGDOUBLE) ? \
-		16 : MAX(sizeof(double), sizeof(long long)))
+#define __CBPRINTF_PACKAGE_ALIGNMENT 16
 #else
-#define CBPRINTF_PACKAGE_ALIGNMENT \
+#define __CBPRINTF_PACKAGE_ALIGNMENT \
 	(IS_ENABLED(CONFIG_CBPRINTF_PACKAGE_LONGDOUBLE) ? \
 		sizeof(long double) : MAX(sizeof(double), sizeof(long long)))
 #endif
 
+/** @brief Required alignment of the buffer used for packaging. */
+#ifdef Z_POW2_CEIL
```

Comment:
Why is this required? If the toolchain abstraction layer defines `Z_POW2_CEIL`, it should be available for all supported toolchains.

Reply:
I removed this and rebased, but now I can't seem to get the tests to pass anymore, will keep debugging...